Platform/Apple/virtual/src/font/fontEngine.s

```diff
@@ -394,50 +392,43 @@ DoAgn	PHA
 	STA (zTmp1),Y	;write to HGR. Use indrct, indxd adrssing
 	LDA Flg2nd	;check if pixel pattern crosses 2-bytes
 	BEQ Chk8xcp	;if not, then skip to next line of bitmap
-	LDA #0
-	STA Flg2nd	;else, first, clear the flag
+	STY Flg2nd	;else, first, clear the flag
 	LDA Byt2nd	;get the 2nd byte
 	STA zTmp3	;store it in pixel pattern to be plotted
 	LDA MskBytH
 	INY  		;increment the byte offset index
-	JMP DoAgn	;go plot the 2nd half of the pixel pattern
-Chk8xcp	LDA Flg8xcp
+	BNE DoAgn	;go plot the 2nd half of the pixel pattern
+Chk8xcp	CMP InvTx_Flg	;save carry for later
+	LDA Flg8xcp
 	BEQ SkpLine
 	INY
-	LDA InvTx_Flg
-	BNE Chk8xcI
 	LDA (zTmp1),Y
+	BCC Chk8xcI	;CMP was non-zero
 	AND #$FE
 	STA (zTmp1),Y
-	JMP SkpLine
-Chk8xcI	LDA (zTmp1),Y
-	ORA #1
+	BCS SkpLine
+Chk8xcI	ORA #1
 	STA (zTmp1),Y
-	JMP SkpLine
+	BNE SkpLine
 
-NoMask	LDY #0  	;clear the byte offset index
-DoAgnNM	LDA (zTmp1),Y	;get HGR pixels
-	PHA
-	LDA FlgBchr
+NoMask	;;LDY #0  	;clear the byte offset index
+DoAgnNM	LDA FlgBchr
 	BEQ NoBchrP
 	LDA zTmp3
 	EOR #$FF
+	AND (zTmp1),Y	;get HGR pixels
 	STA zTmp3
-	PLA
-	AND zTmp3
-	JMP NoBchrQ
-NoBchrP	PLA
+NoBchrP	LDA (zTmp1),Y	;get HGR pixels
 	ORA zTmp3	;add the char BMP bits into the pixels
-NoBchrQ	ORA #$80	;   (set high bit for the demo)
+	ORA #$80	;   (set high bit for the demo)
 	STA (zTmp1),Y	;write to HGR. Use indrct, indxd adrssing
 	LDA Flg2nd	;check if pixel pattern crosses 2-bytes
 	BEQ SkpLine	;if not, then skip to next line of bitmap
-	LDA #0
-	STA Flg2nd	;else, first, clear the flag
+	STY Flg2nd	;else, first, clear the flag
```

Comment:
Same comment here: unless I'm mistaken, Y is not guaranteed to be zero at this point, so the proposed change will not reliably clear `Flg2nd`.